astropy/io/votable/tree.py

```diff
@@ -2626,6 +2626,12 @@ def _parse_binary(self, mode, iterator, colnumbers, fields, config, pos):
                     mask_bits = careful_read(int((len(fields) + 7) / 8))
                     row_mask_data = list(converters.bitarray_to_bool(
                         mask_bits, len(fields)))
+
+                    # Ignore the mask for string columns (see issue 8995)
+                    for i, f in enumerate(fields):
+                        if f.datatype == 'char' or f.datatype == 'unicodeChar':
+                            row_mask_data[i] = False
```

Comment:
Do you think we should emit warning when this happens?

Reply:
I don't think so.  For the data consumer, they are getting the behavior they expect for empty strings (the behavior they would get from the other serializations), so a warning would be confusing.  

For the data provider, there is a better case for the warning, since setting the null bit for the string may not have meant what they thought.  But the spec is pretty vague on this, so I'd like to get more clarification before issuing a warning.  Two other VOTable validators I tried do not issue a message of any kind for this case.